flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java

```diff
@@ -58,13 +59,19 @@
 
   // Can not just use "type", it conflicts with CATALOG_TYPE.
   public static final String ICEBERG_CATALOG_TYPE = "catalog-type";
+  public static final String ICEBERG_CATALOG_TYPE_HADOOP = "hadoop";
+  public static final String ICEBERG_CATALOG_TYPE_HIVE = "hive";
+  public static final String ICEBERG_CATALOG_TYPE_CUSTOM = "custom";
```

Comment:
Instead of using `type=custom` and `impl=com.example.Catalog`, why not just combine them into `type=com.example.Catalog`. We can try to load the type as an implementation class if it isn't a well-known name like "hive".

Reply:
@aokolnychyi and @RussellSpitzer, do you have an opinion here?

Reply:
Yeah this sounds like a cleaner way to go, the only disadvantage is that we are overloading the term `type`. I have updated the code.